packages/clerk-js/src/ui/signUp/utils.ts

```diff
@@ -45,5 +70,75 @@ export function determineFirstPartyFields(environment: EnvironmentResource, hasT
     fields.password = 'required';
   }
 
+  // Enable fields based on current form state
+
+  const shouldShowEmailAddressField =
+    (isEmailAddressUsedForIdentification && hasTicket && hasEmail) ||
+    fields.emailAddress ||
+    (fields.emailOrPhone && activeCommIdentifierType === 'emailAddress');
+
+  if (shouldShowEmailAddressField) {
+    fields.emailAddress ||= 'required'; // Is this accurate?
```

Comment:
🔧 These are remnants of the old auth config modelling. Currently we have the user_settings object and it's better to build a modelling on top of it without using hardcoded enumerations.

Reply:
The field determination function was given a complete overhaul, please take a look.